proxy/envoy/config.go

```diff
@@ -300,7 +300,31 @@ func buildOutboundHTTPRoutes(
 				http.VirtualHosts = append(http.VirtualHosts, host)
 				clusters = append(clusters, host.clusters()...)
 
-			case model.ProtocolTCP, model.ProtocolHTTPS:
+			case model.ProtocolHTTPS:
+				if service.External() {
+					routes := make([]*HTTPRoute, 0)
+
+					cluster := buildOutboundCluster(service.Hostname, servicePort, nil)
```

Comment:
Can you make this a function `useEgressCluster(route)` on a default route and reduce code duplication with the above?